client/state/themes/selectors.js

```diff
@@ -410,17 +410,20 @@ export function getThemeHelpUrl( state, themeId, siteId ) {
 
 /**
  * Returns the URL for purchasing the given theme for the given site.
+ * If the site is a Jetpack site, return the plans URL.
  *
  * @param  {Object}  state   Global state tree
  * @param  {String}  themeId Theme ID
  * @param  {Number}  siteId  Site ID for which to buy the theme
  * @return {?String}         Theme purchase URL
  */
 export function getThemePurchaseUrl( state, themeId, siteId ) {
-	if ( isJetpackSite( state, siteId ) || ! isThemePremium( state, themeId ) ) {
+	if ( ! isThemePremium( state, themeId ) ) {
 		return null;
 	}
-
+	if ( config.isEnabled( 'jetpack/pijp' ) && isJetpackSite( state, siteId ) ) {
+		return `/plans/${ getSiteSlug( state, siteId ) }`;
```

Comment:
Would you mind moving this to a separate selector? I think it's good to keep them distinct, since what they do is rather different (and is mirrored by the different `theme-options` now).